packages/react-instantsearch-dom/src/components/__tests__/VoiceSearch.tsx

```diff
@@ -0,0 +1,116 @@
+import React from 'react';
+import renderer from 'react-test-renderer';
+import Enzyme, { mount } from 'enzyme';
+import Adapter from 'enzyme-adapter-react-16';
+import VoiceSearch from '../VoiceSearch';
+
+Enzyme.configure({ adapter: new Adapter() });
+
+const renderAndMatchSnapshot = (element: JSX.Element) => {
```

Comment:
Is this function wrapper necessary? In latest tests I think we use Enzyme for this. There's less boilerplate.

Reply:
@francoischalifour 
Oh the latest way is just using Enzyme for matching snapshot. I must've followed some old way doing it. Anyway, I have a question on this.

```js
      const wrapper = shallow(<VoiceSearch {...defaultProps} />);
      expect(wrapper).toMatchSnapshot();
```

This produced

```js
exports[`VoiceSearch Rendering with default props 1`] = `
<VoiceSearch
  isBrowserSupported={true}
  isListening={false}
  toggleListening={[Function]}
  translate={[Function]}
  voiceListeningState={
    Object {
      "errorCode": undefined,
      "isSpeechFinal": undefined,
      "status": "initial",
      "transcript": undefined,
    }
  }
/>
`;
```

Do you have any idea why it's not actual dom elements?

Reply:
Probably because of the serializer: https://github.com/algolia/react-instantsearch/blob/c02cd328e80066c8d3799beed714bb7cd0553153/jest.config.js#L5

Reply:
Hmm, but I don't understand why 

```js
const wrapper = shallow(<QueryRuleCustomData {...props} />);
...
expect(wrapper).toMatchSnapshot();
```

generates 

```js
exports[`QueryRuleCustomData expects to render the empty container with empty items 1`] = `
<div
  className="ais-QueryRuleCustomData"
/>
`;
```

According to the serializer, the result should've been something like this

```js
exports[`QueryRuleCustomData expects to render the empty container with empty items 1`] = `
<QueryRuleCustomData />
`;
```